complex/ft_domain.c

```diff
@@ -206,6 +205,36 @@ static int ft_setup_xcontrol_bufs(struct ft_xcontrol *ctrl)
 	return 0;
 }
 
+static int ft_setup_mr_control(struct ft_mr_control *ctrl)
+{
+	size_t size;
+	int ret;
+	uint64_t access;
+
+	size = ft_ctrl.size_array[ft_ctrl.size_cnt - 1];
+	if (!ctrl->buf) {
+		ctrl->buf = malloc(size);
+		if (!ctrl->buf)
+			return -FI_ENOMEM;
+	} else {
+		memset(ctrl->buf, 0, size);
+	}
```

Comment:
The memset likely isn't needed (please confirm with where it is allocated).  Also, I would convert the malloc() above to calloc(), so that the buffer is zeroed.